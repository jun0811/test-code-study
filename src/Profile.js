import React from "react";

export const Profile = ({ username, name }) => {
  return (
    <div>
      <b>{username}</b>&nbsp;
      <span>({name})</span>
    </div>
  );
};
