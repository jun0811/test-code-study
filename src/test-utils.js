import React, { ReactElement } from "react";
import { render } from "@testing-library/react";

// const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
//   return (
//   );
// };

const customRender = (ui, options) => render(ui, { ...options });

export * from "@testing-library/react";
export { customRender as render };
