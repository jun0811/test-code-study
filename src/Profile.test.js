import { render, screen } from "test-utils";
import { Profile } from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="lsj" name="이승준" />);
    expect(utils.container).toMatchSnapshot();
  });

  it("shows the props correctly", () => {
    render(<Profile username="lsj" name="이승준" />);

    screen.getByText("lsj"); // velopert 라는 텍스트를 가진 엘리먼트가 있는지 확인
    screen.getByText("(이승준)"); // (김민준) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    screen.getByText(/이/); // 정규식 /김/ 을 통과하는 엘리먼트가 있는지 확인
  });
});
