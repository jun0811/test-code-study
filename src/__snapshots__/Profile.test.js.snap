// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      lsj
    </b>
     
    <span>
      (
      이승준
      )
    </span>
  </div>
</div>
`;
